persistence/state.go

```diff
@@ -0,0 +1,187 @@
+package persistence
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"log"
+
+	"github.com/celestiaorg/smt"
+	"github.com/pokt-network/pocket/persistence/types"
+	"google.golang.org/protobuf/proto"
+)
+
+type MerkleTree float64
+
+// A work-in-progress list of all the trees we need to update to maintain the overall state
+const (
+	// Actor  Merkle Trees
+	appMerkleTree MerkleTree = iota
+	valMerkleTree
+	fishMerkleTree
+	serviceNodeMerkleTree
+
+	// Account Merkle Trees
+	accountMerkleTree
+	poolMerkleTree
+
+	// Data / State Merkle Trees
+	blocksMerkleTree
+	paramsMerkleTree
+	flagsMerkleTree
+
+	// Used for iteration purposes only - see https://stackoverflow.com/a/64178235/768439
+	lastMerkleTree
```

Comment:
`Limit` is more of an appropriate naming convention - this sounds like the `last block merkle tree`